### PR TITLE
Us 9 add index pages

### PR DESCRIPTION
**Pull request by dpcodes**
Just so we can discuss if this is the structure you'd like, or if you have any suggestion on how we should build it.

Kinda figuring things out as I go, I have limited vue/nuxt experience.

I went for the layout + page recipe

- `JobHeader` will be on every job page, the image, background image and job name will be dynamic based on the route.
- The link in the swiper will redirect to the appropriate `/role/job` route

Should probably also consider setting up a code formatter so that we don't have more changes than necessary in the prs. I usually use prettier, but biome is also very nice out of the box and includes a linter.

### CODE REVIEW

File: components/JobHeader.vue [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<template>
+  <div
+    :class="`bg-${job} w-full -z-1 bg-top bg-cover absolute top-0 h-64 md:h-96`"
```
Comment (sneakycrow) [inline components/JobHeader.vue:3]: I presume it's this line that's not working? This is a tricky tailwind conundrum.

I haven't looked deeply into the tailwind docs for this, but the conclusion I've run into is if you try to add a dynamic class, particularly one with string concatenation, tailwind won't pick it up and therefore won't add the styles to the css.

It's annoyingly more verbose, but the approach I took to solve this is in the RoleSlider.vue component. For the keys, we use the full class we dynamically apply and then the right is just a boolean value of wether to enable that class.

https://github.com/sneakycrow/balance-xiv/blob/d5b0be0ea10a0d206c140fd16e5ebbe6360f1a0d/components/RoleSlider.vue#L12-L16

I'm very open to alternative ways to do this, but this accurately applies the styles. It's just annoyingly verbose. I wonder if there's a more "tailwind-y" way to do this. Or, if not, maybe we add a utility function for appropriately returning the job/role classes

File: models/PartialJob.ts
Comment (sneakycrow) [inline models/PartialJob.ts:1]: Nice job moving this to its own type! I think we can remove the `Partial` from the name (for Role as well)

File: pages/tanks/[job].vue [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<template>
+  <div></div>
+</template>
+
+<script setup lang="ts">
+definePageMeta({
```
Comment (sneakycrow) [inline pages/tanks/[job].vue:6]: Oooh clean! I didn't know about this function 😄

File: components/RoleSlider.vue [hunk 3]
```diff
@@ -92,20 +93,7 @@ export default {
...
-            description: string;
-            name: string;
-            nameSingular: string;
-        }
-        type PartialJob = {
-            slug: string;
-            name: string;
-            description: string;
-        }
+
```
Comment (sneakycrow) [inline components/RoleSlider.vue:96]: Thanks for cleaning these up!

File: components/JobHeader.vue (hunk 1 above)
Comment (dpcodes) [reply components/JobHeader.vue:3]: I did try with `bg-paladin` directly and the styles were still not applying. 

Also, as I read your comment, it started working with `bg-${job}` so I am very confused 😅

Comment (sneakycrow) [reply components/JobHeader.vue:3]: That's interesting! Well jeez if it works I say send it

**Comment by sneakycrow** [comment]
> Should probably also consider setting up a code formatter so that we don't have more changes than necessary in the prs. I usually use prettier, but biome is also very nice out of the box and includes a linter.

Yes! I totally agree. I went ahead and merged [this](#15) per your suggestion.

It did cause a few merge conflicts in this PR, but both _should_ be fairly minor. If they end up being a PITA let me know and I'll fix them up for you

**Comment by dpcodes** [comment]
@sneakycrow I moved the assets to the public folder. Every job should have a landing page with the header dynamically changing icon and job name

**Review by sneakycrow — APPROVED** [review]
Awesome changes, thank you!

**Review by sneakycrow — APPROVED** [review]

File: components/JobHeader.vue [hunk 4]
```diff
@@ -31,7 +31,7 @@
   />
   <div class="w-full h-64 md:h-96 z-10 pl-32 sm:pl-48 lg:pl-60 pt-16 md:pt-24">
     <div class="flex items-center">
-      <img
+      <NuxtImg
```
Comment (sneakycrow) [inline components/JobHeader.vue:34]: Great catch on this one!

File: components/JobHeader.vue (hunk 1 above)
Comment (dpcodes) [reply components/JobHeader.vue]: Sorry to necro this, i've read a bit more about it and the issue comes from the fact that tailwind doesn't have any type of client-side runtime. This means that the class needs to exist at build time for it to be picked up. An alternative they suggest is directly putting purely dynamic values in the style tag, which i think could make sense in this case and remove this whole conditional and verbose object. Let me know what you think 🤔

Comment (sneakycrow) [reply components/JobHeader.vue]: > An alternative they suggest is directly putting purely dynamic values in the style tag

I'm not sure I totally understand, could you provide an example or a link?

**Pull request merged**